src/model/training.ts

```diff
@@ -0,0 +1,22 @@
+import {Student} from "./student";
+import {Trainer} from "./trainer";
+
+export class Training {
+  id : number;
+  name : string;
+  isActive : boolean;
+  start : Date;
+  end : Date;
+  trainers : Array<Trainer>;
+  students : Array<Student>;
```

Comment:
I like this idea, by doing this we can just get in one step from the API what is needed to edit a given training.